scipy/special/tests/test_mpmath.py

```diff
@@ -122,6 +122,30 @@ def test_hyp2f1_strange_points():
 
     FuncData(sc.hyp2f1, dataset, (0,1,2,3), 4, rtol=1e-10).check()
 
+@check_version(mpmath,'1.0.0')
+def test_hyp2f1_integer_c():
+    # a,b,c
+    pts = [
+        (1.35, 2.39, -2),
+        (1.20, 3.82, 0),
+        (-1, 3.2, -1),
+        (1.23, -4, -4),
+        (2.12, 0, -3),
+        (-2, 0.89, -4)
+        ]
+
+    # Test both the complex and real versions of hyp2f1
+    for z in (0.5+0.5j,0.5):
+        curr_pts=[p+(z,) for p in pts]
+        curr_type=np.float if np.isreal(z) else np.complex
+        
+        mp_values=[curr_type(mpmath.hyp2f1(*p)) for p in curr_pts]
+        mp_values=np.array(mp_values, dtype=curr_type)
+
+        scipy_values=[sc.hyp2f1(*p) for p in curr_pts]
+        scipy_values=np.array(scipy_values, dtype=curr_type)
+
+        assert_allclose(scipy_values,mp_values,rtol=1.e-10)
```

Comment:
Look at the uses of `FuncData` elsewhere in the module; probably best to use the same pattern here. (You will get better formatted error reporting, which is always nice.)

Reply:
I tried doing that, but the problem was that FuncData requires a single array of a *single* type in order to evaluate the function. In the case of hyp2f1, I would need to pass in an array that contains both real numbers (a,b,c) and complex numbers (z,mpmath's value). (Note that I can't just pass in a single array of complex numbers since this array can't be passed to hyp2f1 since it doesn't match hyp2f1's signature of real,real,real,complex).

Now I **can** pass a single array of real numbers (basically the columns=a,b,c,z_real,z_imag,expected_real,expected_imag) and pass in a wrapper function to FuncData that converts this array into something that can be passed to hyp2f1.  This is what I did originally did with my text file of 4000 or so test cases.  But I was just afraid that people might think that was overkill for just checking 12 or so values....

So let me know what you prefer.  I already have code that uses FuncData; it's just a matter of cut and paste...

Reply:
I vote FuncData, whenever `assert_allclose` fails you have to go hack up the test to figure out what actually went wrong. The other option (probably more common in the tests) is to use the columns `a, b, c, z, expected` where you make `a`, `b`, and `c` complex and write a wrapper that does `hyp2f1(a.real, b.real, c.real, z)`.